core/model/src/main/java/org/eclipse/rdf4j/model/util/Literals.java

```diff
@@ -502,20 +501,49 @@ public static boolean isLanguageLiteral(Literal literal) {
 
 	/**
 	 * Normalizes the given <a href="https://tools.ietf.org/html/bcp47">BCP47</a> language tag according to the rules
-	 * defined by the JDK in the {@link Locale} API.
+	 * defined in <a href="https://www.rfc-editor.org/rfc/rfc5646.html#section-2.1.1">RFC 5646, section 2.1.1</a>:
+	 * <p>
+	 * <blockquote> All subtags, including extension and private use subtags, use lowercase letters with two exceptions:
+	 * two-letter and four-letter subtags that neither appear at the start of the tag nor occur after singletons. Such
+	 * two-letter subtags are all uppercase (as in the tags "en-CA-x-ca" or "sgn-BE-FR") and four- letter subtags are
+	 * titlecase (as in the tag "az-Latn-x-latn"). </blockquote>
 	 *
 	 * @param languageTag An unnormalized, valid, language tag
 	 * @return A normalized version of the given language tag
 	 * @throws IllformedLocaleException If the given language tag is ill-formed according to the rules specified in
-	 *                                  BCP47.
+	 *                                  BCP47 (RFC 5646).
 	 */
 	public static String normalizeLanguageTag(String languageTag) throws IllformedLocaleException {
-		return new Locale.Builder().setLanguageTag(languageTag).build().toLanguageTag().intern();
+		// check if language tag is well-formed
+		new Locale.Builder().setLanguageTag(languageTag);
```

Comment:
I wonder if we should get rid of this check here, so that we can do (syntactic) as completely separate from verifying the wellformedness of the tag.